### DISCUSSION TITLE
How on earth do I open the tree with a keybind

### DISCUSSION THREAD

**Discussion by ElBread3** [nvim-tree/nvim-tree.lua#2322] [Q&A] [answered]
Simply adding a key bind and nothing happens when I press F2 
what could I be messing up 

local function my_on_attach(bufnr)
  local api = require "nvim-tree.api"

  local function opts(desc)
    return { desc = "nvim-tree: " .. desc, buffer = bufnr, noremap = true, silent = true, nowait = true }
  end

  -- default mappings
  api.config.mappings.default_on_attach(bufnr)

  vim.keymap.set('n', 'O', '', { buffer = bufnr })
  vim.keymap.del('n', 'O', { buffer = bufnr })
  vim.keymap.set('n', '<2-RightMouse>', '', { buffer = bufnr })
  vim.keymap.del('n', '<2-RightMouse>', { buffer = bufnr })
  vim.keymap.set('n', 'D', '', { buffer = bufnr })
  vim.keymap.del('n', 'D', { buffer = bufnr })
  vim.keymap.set('n', 'E', '', { buffer = bufnr })
  vim.keymap.del('n', 'E', { buffer = bufnr })

  vim.keymap.set('n', 'A', api.tree.expand_all, opts('Expand All'))
  vim.keymap.set('n', '?', api.tree.toggle_help, opts('Help'))
  vim.keymap.set('n', 'C', api.tree.change_root_to_node, opts('CD'))
  
  vim.keymap.set('n', '<F2>', api.tree.toggle, opts('Toggle Tree'))
end

-- pass to setup along with your other options
require("nvim-tree").setup {
  ---
  on_attach = my_on_attach,
  ---
}

to specify, I'm using Windows

**Comment by gegoune** [comment, marked as answer]
You didn't show mapping for F2, so not sure what it is suppose to do, but you will need to set it up outside of mvim-tree's configuration. Keymaps that you are setting above are only to be active within nvim-tree buffer.

**Reply by ElBread3** [reply]
Thanks! That's a great clarification. I can now open it with
vim.keymap.set('n', '<S-n>', "<cmd> NvimTreeToggle <CR>")
as the shift key seems mostly unused
